components/tidb_query_datatype/src/codec/mysql/duration.rs

```diff
@@ -204,20 +204,46 @@ mod parser {
         Ok((rest, frac * TEN_POW[NANO_WIDTH.saturating_sub(len)]))
     }
 
+    /// a string can match datetime format only if it starts with a series of digits
+    /// whose length matches the full format of DateTime literal (12, 14)
+    /// or the string start with a date literal
+    fn format_can_match_datetime(input: &str) -> IResult<(), (), ()> {
```

Comment:
Seems this function does the exact same logic judgment as `hhmmss_datetime`. Is it really necessary?

Reply:
This function serves as an exact counterpart of [tidb::types::format_can_match_datetime](https://github.com/pingcap/tidb/blob/a33d971adde7e5750cbddb2db2664b3389b0ab69/types/time.go#L1748). And if we use hhmmss_datetime instead to check if the string matches the datetime format, there might be some other cases that will fail the check.

Reply:
Got it.